service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/Twin.java

```diff
@@ -109,29 +108,35 @@ public static Twin fromJson(String json)
         }
 
         // Desired properties
-        twin.getDesiredProperties().setVersion(twinState.getDesiredProperties().getVersion());
-        if (twinState.getDesiredProperties().size() > 0)
-        {
-            twin.getDesiredProperties().putAll(twinState.getDesiredProperties());
-        }
-
-        twin.getDesiredProperties().setTwinMetadata(twinState.getDesiredProperties().getTwinMetadata());
-        if (twinState.getDesiredProperties().getMetadataMap().size() > 0)
+        if (twinState.getDesiredProperties() != null)
         {
-            twin.getDesiredProperties().getMetadataMap().putAll(twinState.getDesiredProperties().getMetadataMap());
+            twin.getDesiredProperties().setVersion(twinState.getDesiredProperties().getVersion());
+            if (twinState.getDesiredProperties().size() > 0)
+            {
+                twin.getDesiredProperties().putAll(twinState.getDesiredProperties());
+            }
+
+            twin.getDesiredProperties().setTwinMetadata(twinState.getDesiredProperties().getTwinMetadata());
+            if (twinState.getDesiredProperties().getMetadataMap().size() > 0)
+            {
+                twin.getDesiredProperties().getMetadataMap().putAll(twinState.getDesiredProperties().getMetadataMap());
+            }
         }
 
-        // Reported properties
-        twin.getReportedProperties().setVersion(twinState.getReportedProperties().getVersion());
-        if (twinState.getReportedProperties().size() > 0)
+        if (twinState.getReportedProperties() != null)
         {
-            twin.getReportedProperties().putAll(twinState.getReportedProperties());
-        }
-
-        twin.getReportedProperties().setTwinMetadata(twinState.getReportedProperties().getTwinMetadata());
-        if (twinState.getReportedProperties().getMetadataMap().size() > 0)
-        {
-            twin.getReportedProperties().getMetadataMap().putAll(twinState.getReportedProperties().getMetadataMap());
+            // Reported properties
```

Comment:
Maybe move this comment above L126, before we do the null check for reported properties?